llvm/include/llvm/MC/MCContext.h

```diff
@@ -136,6 +136,9 @@ class MCContext {
   /// objects.
   BumpPtrAllocator Allocator;
 
+  /// For MCFragment instances.
+  BumpPtrAllocator FragmentAllocator;
```

Comment:
Any particular reasons to not reuse `Allocator`?

Reply:
Make fragments closer for locality.

In addition, there might be an opportunity to reduce `MCFragment::destroy` (primarily called by `~MCSection`) overhead.
If we can obtain BumpPtrAllocator slabs, there will probably be a better way to implement `-mc-disable-free` like `-cc1 -disable-free`.

Reply:
Good point, ok